pkg/tfgen/source.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfgen
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+)
+
+type DocsSource interface {
+	getResource(rawname string, info tfbridge.ResourceOrDataSourceInfo) (*DocFile, error)
```

Comment:
Is it not strange that a DocSource needs to know `info`? What is it reading from it? I see it's reading `info.GetDocs()` - is it a good idea for DocsSource's job to be merging overrides from info? Consider if you had Git-based and some other say AI-based DocsSource, both would have to implement overrides.